src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -601,4 +607,70 @@ public String getQuestionAdditionalInfoHtml() {
         return getQuestionDetails().getQuestionAdditionalInfoHtml(questionNumber, "");
     }
     
+    public List<FeedbackPathAttributes> getFeedbackPaths(List<FeedbackPath> feedbackPathEntities) {
+        List<FeedbackPathAttributes> feedbackPaths =
+                new ArrayList<FeedbackPathAttributes>();
+        for (FeedbackPath feedbackPath : feedbackPathEntities) {
+            feedbackPaths.add(new FeedbackPathAttributes(feedbackPath));
+        }
+        return feedbackPaths;
+    }
+    
+    public List<FeedbackPath> getFeedbackPathEntities(List<FeedbackPathAttributes> feedbackPaths) {
+        List<FeedbackPath> feedbackPathEntities = new ArrayList<FeedbackPath>();
+        if (feedbackPaths != null) {
+            for (FeedbackPathAttributes feedbackPath : feedbackPaths) {
+                feedbackPathEntities.add(feedbackPath.toEntity());
+            }
+        }
+        
+        return feedbackPathEntities;
+    }
+    
+    public List<FeedbackPath> getFeedbackPathEntities() {
+        return getFeedbackPathEntities(feedbackPaths);
+    }
+    
+    /** 
+     * Verifies whether the given student is a giver in the question's feedback paths
+     */
+    public boolean hasStudentAsGiverInFeedbackPaths(StudentAttributes student) {
+        if (giverType == FeedbackParticipantType.CUSTOM && recipientType == FeedbackParticipantType.CUSTOM) {
+            for (FeedbackPathAttributes feedbackPath : feedbackPaths) {
+                if (feedbackPath.isStudentFeedbackPathGiver(student)) {
+                    return true;
+                }
+            }
+        }
```

Comment:
You can reverse the first `if` condition to reduce nesting. Try to avoid if conditions that swallow a very long chunk of code.